### ISSUE TITLE
Pylint not suppressing the stdout of C loaded extensions

### ISSUE THREAD

**Issue by willpxxr** [PyCQA/pylint#3518]
### Steps to reproduce
1. Pylint configured to load a module which contains a C extension ie. Pygame, and report in JSON

```ini
[MASTER]
extension-pkg-whitelist=pygame
...
[REPORTS]
output-format=json
```

2. Run Pylint on a file which contains a call to the method containing a C extension.

```python
import pygame
if __name__ == "__main__":
        pygame.init()
```
```bash
pylint main.py > output.json
```
3. Cat the output and the result should look as follows:
```json
pygame 1.9.6
Hello from the pygame community. https://www.pygame.org/contribute.html
[
    {
        "type": "convention",
        "module": "settings",
        "obj": "",
        "line": 1,
        "column": 0,
        "path": "src/games/space_invaders/settings.py",
        "symbol": "missing-module-docstring",
        "message": "Missing module docstring",
        "message-id": "C0114"
    },
...
```

### Current behavior
Currently Pylint is not suppressing messages written to the stdout of C extensions which get loaded. This results in a report generation which does not follow the json formatting, as it contains anything which may be printed to the stdout by a C extension during analysis.

### Expected behavior
Pylint shouldn't print the stdout messages of C extensions being ran during analysis. 
ie the `report.json` should look as follows:
```json
[
    {
        "type": "convention",
        "module": "settings",
        "obj": "",
        "line": 1,
        "column": 0,
        "path": "src/games/space_invaders/settings.py",
        "symbol": "missing-module-docstring",
        "message": "Missing module docstring",
        "message-id": "C0114"
    },
...
```

### pylint --version output
```bash
circleci@b0feadac759a:~$ pylint --version
pylint 2.4.4
astroid 2.3.3
Python 3.6.10 (default, Apr 21 2020, 14:40:51)
[GCC 8.3.0]

```

**Comment by kuroi-okami** [comment]
Attached my [pylintrc](https://github.com/PyCQA/pylint/files/4536605/pylintrc.zip) for reference

**Comment by PCManticore** [comment]
Thanks for the report, definitely makes sense.

**Comment by piotr-machura** [comment]
Hi, this issue causes problems when using pylint's json format in language servers - in the case of pygame it can be ad-hoc fixed by adding `export PYGAME_HIDE_SUPPORT_PROMPT="hide"` to the working environment, but it would be great to have a proper fix for other such C extensions.
